### PR TITLE
added test files for sql

### PR DESCRIPTION
**Pull request by sharadregoti**
Code coverage : 34.3%
some bug in the program, not able to find out.
test passes most of the time then test fails relacing string1 with string2 for update and delete test files

### CODE REVIEW

File: modules/crud/sql/delete_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"github.com/spaceuptech/space-cloud/model"
+)
+
+func TestGenerateDeleteQuery(t *testing.T) {
+	// truecases := 11
+	tests := []struct {
+		name, tableName, want string
+		req                   model.DeleteRequest
+	}{
+		{name: "Successfull Test", tableName: "fooTable", want: "DELETE FROM fooTable WHERE ((String1 = ?) AND (String2 = ?))", req: model.DeleteRequest{Find: map[string]interface{}{"String1": "1", "String2": "2"}}},
```
Comment (sharadregoti) [inline modules/crud/sql/delete_test.go]: output should be this : "DELETE FROM fooTable WHERE ((String1 = ?) AND (String2 = ?))"
but sometimes string1 & string2 interchange
"DELETE FROM fooTable WHERE ((String2 = ?) AND (String1 = ?))"

File: modules/crud/sql/update_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+func TestGenerateUpdateQuery(t *testing.T) {
+	truecases := 10
+	var ctx context.Context
+	project := "projectName"
+	tests := []struct {
+		name, tableName, want string
+		req                   model.UpdateRequest
+	}{
+		{name: "Successfull Test", tableName: "fooTable", want: "UPDATE fooTable SET String1=? WHERE ((FindString1 = ?) AND (FindString2 = ?))", req: model.UpdateRequest{Update: map[string]interface{}{"$set": map[string]interface{}{"String1": "1"}}, Find: map[string]interface{}{"FindString1": "1", "FindString2": "2"}}},
```
Comment (sharadregoti) [inline modules/crud/sql/update_test.go]: same here

File: modules/crud/sql/delete_test.go (hunk 1 above)
Comment (Jayesh333) [reply modules/crud/sql/delete_test.go]: The where clause parameters may get interchanged. Write your test cases to accommodate both the results using the or

File: modules/crud/sql/update_test.go (hunk 2 above)
Comment (Jayesh333) [reply modules/crud/sql/update_test.go]: Same

File: modules/crud/sql/create.go [hunk 3]
```diff
@@ -16,15 +16,15 @@ import (
 
 // Create inserts a document (or multiple when op is "all") into the database
 func (s *SQL) Create(ctx context.Context, project, col string, req *model.CreateRequest) error {
-	sqlQuery, args, err := s.generateCreateQuery(ctx, project, col, req)
+	sqlQuery, args, err := s.GenerateCreateQuery(ctx, project, col, req)
```
Comment (Jayesh333) [inline modules/crud/sql/create.go]: Don't export this function

File: modules/crud/sql/create.go [hunk 4]
```diff
@@ -16,15 +16,15 @@ import (
...
 	if err != nil {
 		return err
 	}
 	return s.doExec(sqlQuery, args)
 }
 
 //generateCreateQuery makes query for create operation
-func (s *SQL) generateCreateQuery(ctx context.Context, project, col string, req *model.CreateRequest) (string, []interface{}, error) {
+func (s *SQL) GenerateCreateQuery(ctx context.Context, project, col string, req *model.CreateRequest) (string, []interface{}, error) {
```
Comment (Jayesh333) [inline modules/crud/sql/create.go]: don't export this function

File: modules/crud/sql/delete.go [hunk 5]
```diff
@@ -14,15 +14,15 @@ import (
...
+	sqlString, args, err := s.GenerateDeleteQuery(ctx, project, col, req)
 	if err != nil {
 		return err
 	}
 	return s.doExec(sqlString, args)
 }
 
 //genrateDeleteQuery makes query for delete operation
-func (s *SQL) generateDeleteQuery(ctx context.Context, project, col string, req *model.DeleteRequest) (string, []interface{}, error) {
+func (s *SQL) GenerateDeleteQuery(ctx context.Context, project, col string, req *model.DeleteRequest) (string, []interface{}, error) {
```
Comment (Jayesh333) [inline modules/crud/sql/delete.go]: don't export this function

File: modules/crud/sql/delete_test.go (hunk 1 above)
Comment (Jayesh333) [reply modules/crud/sql/delete_test.go]: put an or clause. both the results are right

File: modules/crud/sql/update_test.go (hunk 2 above)
Comment (Jayesh333) [reply modules/crud/sql/update_test.go]: both results are right

File: modules/crud/sql/create.go (hunk 3 above)
Comment (sharadregoti) [reply modules/crud/sql/create.go]: I tried to not export these func
But while running go test command
It says not test files found or undefined function name

Comment (YourTechBud) [reply modules/crud/sql/create.go]: so the name of the testing function in *_test.go will always start with "Test". So the test functions will always be exported but the function to be tested can be un-exported.

**Pull request closed without merging**